src/tbx/mod.rs

```diff
@@ -0,0 +1,418 @@
+/// building indices and converting BCF to VCF text output.  Both is out of scope here.
+#[derive(Debug)]
+pub struct Reader {
+    /// The header lines (if any).
+    header: Vec<String>,
+
+    /// The file to read from.
+    hts_file: *mut htslib::htsFile,
+    /// The file format information.
+    hts_format: htslib::htsExactFormat,
+    /// The tbx_t structure to read from.
+    tbx: *mut htslib::tbx_t,
+    /// The current buffer.
+    buf: htslib::kstring_t,
+    /// Iterator over the buffer.
+    itr: Option<*mut htslib::hts_itr_t>,
+
+    /// The currently fetch region's tid.
+    tid: i32,
+    /// The currently fetch region's 0-based begin pos.
+    start: i32,
+    /// The currently fetch region's 0-based end pos.
+    end: i32,
+}
+
+unsafe impl Send for Reader {}
+
+/// Redefinition of `KS_SEP_LINE` from `htslib/kseq.h`.
+const KS_SEP_LINE: i32 = 2;
+
+impl Reader {
+    /// Create a new Reader from path.
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - the path to open.
+    pub fn from_path<P: AsRef<Path>>(path: P) -> Result<Self, TabixReaderPathError> {
+        if let Some(p) = path.as_ref().to_str() {
+            Ok(try!(Self::new(p.as_bytes())))
+        } else {
+            Err(TabixReaderPathError::InvalidPath)
+        }
+    }
+
+    pub fn from_url(url: &Url) -> Result<Self, TabixReaderError> {
+        Self::new(url.as_str().as_bytes())
+    }
+
+    /// Create a new Reader.
+    ///
+    /// # Arguments
+    ///
+    /// * `path` - the path.
+    fn new(path: &[u8]) -> Result<Self, TabixReaderError> {
+        let path = ffi::CString::new(path).unwrap();
+        let hts_file =
+            unsafe { htslib::hts_open(path.as_ptr(), ffi::CString::new("r").unwrap().as_ptr()) };
+        let hts_format = unsafe { (*htslib::hts_get_format(hts_file)).format };
+        let tbx = unsafe { htslib::tbx_index_load(path.as_ptr()) };
+        let mut header = Vec::new();
+        let mut buf = htslib::kstring_t {
+            l: 0,
+            m: 0,
+            s: ptr::null_mut(),
+        };
+        unsafe {
+            while htslib::hts_getline(hts_file, KS_SEP_LINE, &mut buf) >= 0 {
+                if buf.l > 0 && (*buf.s) as i32 == (*tbx).conf.meta_char {
+                    header.push(String::from(ffi::CStr::from_ptr(buf.s).to_str().unwrap()));
+                } else {
+                    break;
+                }
+            }
+        }
+
+        if tbx.is_null() {
+            Err(TabixReaderError::InvalidIndex)
+        } else {
+            Ok(Reader {
+                header,
+                hts_file,
+                hts_format,
+                tbx,
+                buf,
+                itr: None,
+                tid: -1,
+                start: -1,
+                end: -1,
+            })
+        }
+    }
+
+    /// Get sequence ID from sequence name.
+    pub fn seq_name_to_id(&self, name: &str) -> Result<u32, SequenceLookupError> {
```

Comment:
Maybe `id`, similar to `tid` in the bam `HeaderView`: https://docs.rs/rust-htslib/0.17.0/rust_htslib/bam/struct.HeaderView.html#method.tid

Reply:
OK